crates/blockchain/blockchain.rs

```diff
@@ -1020,7 +1021,7 @@ fn verify_blob_gas_usage(block: &Block, config: &ChainConfig) -> Result<(), Chai
     if block
         .header
         .blob_gas_used
-        .is_some_and(|header_blob_gas_used| header_blob_gas_used != blob_gas_used)
+        .is_some_and(|header_blob_gas_used| header_blob_gas_used != blob_gas_used as u64)
```

Comment:
This seems dangerous. If the header stores a u64 then values greater than u32::MAX are legal, but we would fail for those.